include/RAJA/policy/hip/teams.hpp

```diff
@@ -268,6 +268,48 @@ struct LoopExecute<hip_global_thread<DIM0, DIM1, DIM2>, SEGMENT> {
   }
 };
 
+/*
+Reshape threads in a block into a 1D iteration space
+*/
+struct flatten_block_threads_direct;
+struct flatten_block_threads_loop;
+
+template<typename SEGMENT>
+struct LoopExecute<flatten_block_threads_direct, SEGMENT>
+{
+  template<typename BODY>
+  static RAJA_INLINE RAJA_DEVICE void exec(
+      LaunchContext const RAJA_UNUSED_ARG(&ctx),
+      SEGMENT const &segment,
+      BODY const &body)
+  {
+    const int len = segment.end() - segment.begin();
+    {
+      const int tid = threadIdx.x + blockDim.x*(threadIdx.y + blockDim.y*threadIdx.z);
+
+      if (tid < len) body(*(segment.begin() + tid));
+    }
+  }
+};
+
+template<typename SEGMENT>
+struct LoopExecute<flatten_block_threads_loop, SEGMENT>
+{
+  template<typename BODY>
+  static RAJA_INLINE RAJA_DEVICE void exec(
+      LaunchContext const RAJA_UNUSED_ARG(&ctx),
+      SEGMENT const &segment,
+      BODY const &body)
+  {
+    const int len = segment.end() - segment.begin();    
+    for(int tid = threadIdx.x + blockDim.x*(threadIdx.y + blockDim.y*threadIdx.z);
+        tid < len;
+        tid += blockDim.x * blockDim.y * blockDim.z){
+      body(*(segment.begin() + tid));
+    }
+  }
+};
+
```

Comment:
Don't you need to use the hip vars here? Like `hipThreadIdx_x` instead of `threadIdx.x`? Here's a list: https://rocmdocs.amd.com/en/latest/Programming_Guides/HIP-GUIDE.html#built-in-variables

Reply:
We have been using threadIdx.x in RAJA; they are both compatible.

Reply:
In that case the routines are exactly the same so can we just have one in a common file (or include one file from the other?)

Reply:
Also is the `threadIdx` struct something hip guarantees keeping in their api? I think the answer is probably yes if it's being used in other parts of RAJA

Reply:
Yes!, that having a common file will be a part of a different PR.